File: backend/src/main/java/com/demo/app/dto/ContactRequest.java
package com.demo.app.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

public record ContactRequest(
        @NotBlank @Size(max = 255) String fullName,
        @NotBlank @Size(max = 255) String streetLine1,
        @Size(max = 255) String streetLine2,
        @NotBlank @Size(max = 100) String city,
        @Size(max = 100) String stateProvince,
        @Size(max = 20) String postalCode,
        @NotBlank @Size(max = 100) String country,
        @Size(max = 50) String phone
) {}

File: frontend/app/Http/Requests/ContactRequest.php
<?php

declare(strict_types=1);

namespace App\Http\Requests;

use Illuminate\Foundation\Http\FormRequest;

class ContactRequest extends FormRequest
{
    public function authorize(): bool
    {
        return true;
    }

    public function rules(): array
    {
        return [
            'fullName' => 'required|string|max:255',
            'email' => 'nullable|string|email|max:255',
            'phone' => 'nullable|string|max:255',
            'streetLine1' => 'required|string|max:255',
            'streetLine2' => 'nullable|string|max:255',
            'city' => 'required|string|max:100',
            'country' => 'required|string|max:100',
        ];
    }
}
